core/src/main/java/org/apache/iceberg/util/SortStrategyUtil.java

```diff
@@ -0,0 +1,223 @@
+  private static final Logger LOG = LoggerFactory.getLogger(SortStrategyUtil.class);
+
+  private SortStrategyUtil() {
+  }
+
+  /**
+   * Measures the sortedness score for a given list of {@link DataFile}s.
+   *
+   * @param files List of {@link DataFile}s
+   * @param sortOrder The sort order
+   * @return A score between 0.0 and 1.0 to represent how well those data files are sorted
+   * on the sort order.
+   */
+  public static double sortednessScore(List<DataFile> files, SortOrder sortOrder)  {
+    if (files.size() == 1) {
+      return 1.0;
+    }
+
+    List<Integer> sortColIndices = getSortColumnIndices(sortOrder);
+    if (sortColIndices.isEmpty()) {
+      // Cannot obtain indices of sort columns, will consider those files are completely unsorted
+      return 0.0;
+    }
+
+    double totalScore = 0;
+    // Compute and aggregate scores for all lowerBounds
+    List<List<ByteBuffer>> lowerBounds = getColumnValueBounds(files, sortColIndices, true);
+    for (List<ByteBuffer> point : lowerBounds) {
+      totalScore += overlapScore(point, files, sortColIndices, sortOrder.schema());
+    }
+    // Compute and aggregate scores for all upperBounds
+    List<List<ByteBuffer>> upperBounds = getColumnValueBounds(files, sortColIndices, false);
+    for (List<ByteBuffer> point : upperBounds) {
+      totalScore += overlapScore(point, files, sortColIndices, sortOrder.schema());
+    }
+    // Then normalize the result
+    return totalScore / (2 * files.size());
+  }
+
+  /**
+   * Removes files that are already sorted (having no overlap on sort column values with other files)
+   * from a collection of file scans.
+   *
+   * @param fileScans The original collection of file scans
+   * @param sortOrder The sort order
+   * @return A collection of file scans after removing sorted ones from the original collection
+   */
+  public static Iterable<FileScanTask> removeSortedFiles(Iterable<FileScanTask> fileScans, SortOrder sortOrder) {
+    List<Integer> sortColIndices = getSortColumnIndices(sortOrder);
+    List<DataFile> dataFiles = FluentIterable.from(fileScans).stream()
+        .map(FileScanTask::file)
+        .collect(Collectors.toList());
+    return FluentIterable.from(fileScans)
+        .filter(f -> hasOverLap(f.file(), dataFiles, sortColIndices, sortOrder.schema()));
+  }
+
+  private static boolean hasOverLap(
+      DataFile file,
+      List<DataFile> dataFiles,
+      List<Integer> sortColIndices,
+      Schema sortSchema) {
+    for (DataFile f : dataFiles) {
+      if (!f.equals(file) && isOverlap(f, file, sortColIndices, sortSchema)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  static List<Integer> getSortColumnIndices(SortOrder sortOrder) {
+    List<Integer> sortColIndices = Lists.newArrayList();
+    for (SortField sortField : sortOrder.fields()) {
+      if (!sortField.transform().isIdentity()) {
+        LOG.info("Cannot measure sortedness score on derived sort column {}",
```

Comment:
Why is this the case? Shouldn't the stored metric values just be transformed already?